### DISCUSSION TITLE
Question about the vector invariant form

### DISCUSSION THREAD

**Discussion by iuryt** [CliMA/Oceananigans.jl#2835] [Computational science!] [answered]
I have been discussing with @glwagner  about different ways of damping numerical noise at smaller scales. He mentioned about how the divergence mode can be used for the biharmonic viscosity and also mentioned about vector invariant advection using WENO. 

As explained in #2440 , I understand how damping in vorticity or divergence helps maintain numerical stability of simulations, as the derivated fields concentrate variance in smaller scales and damping horizontal divergence helps to keep up with weird vertical velocities.

However, I am having trouble understanding vector invariant mode and what its significance is, i.e. how it changes the advection scheme and why it helps damping non-physical noise.

Could someone explain this to me?

**Comment by glwagner** [comment, marked as answer]
@simone-silvestri might be able to provide an in-depth explanation.

The "vector invariant" formulation of momentum advection refers to a way of writing the momentum advection term u . grad(u). It is described here:

https://mitgcm.readthedocs.io/en/latest/algorithm/algorithm.html#vector-invariant-momentum-equations

and also here:

https://journals.ametsoc.org/view/journals/mwre/132/12/mwr2823.1.xml

In terms of damping properties, we find that using WENO to discretize the "flux form" momentum equations (u . grad u) dissipates the variance of velocity --- kinetic energy.

For "WENO vector invariant", we use WENO to reconstruct the vertical vorticity at velocity nodes. This operation appears in a C-grid discretization of the vector invariant form of the momentum equations. We find --- empirically --- that this tends to dissipate enstrophy. It also has an effect of kinetic energy, but it does not tend to dissipate it (cf @simone-silvestri). Also, it does not dissipate spurious divergent energy. Therefore, we must add biharmonic damping of the divergent component of momentum, in addition to using WENO construction of vorticity to dissipate enstrophy.
